lighthouse-viewer/app/src/lighthouse-report-viewer.js

```diff
@@ -88,20 +89,37 @@ class LighthouseReportViewer {
 
   /**
    * Attempts to pull gist id from URL and render report from it.
-   * @return {Promise<void>}
    * @private
    */
   _loadFromDeepLink() {
     const params = new URLSearchParams(location.search);
+
     const gistId = params.get('gist');
-    if (!gistId) {
-      return Promise.resolve();
+    const psiurl = params.get('psiurl');
+
+    let loadPromise = null;
```

Comment:
lets early return and start the blur before we do fetching.

```js
    if (!psiurl && !gistId) return;
    this._toggleLoadingBlur(true);
```

Reply:
personally i'd do a `let loadPromise = Promise.resolve()` here.. and then you don't need an `if (loadPromise)` later on..